src/nvim/sign.c

```diff
@@ -1419,9 +1404,7 @@ static void sign_get_placed_in_buf(
         || (sign_id == 0 && lnum == sign->lnum)
         || (lnum == 0 && sign_id == sign->id)
         || (lnum == sign->lnum && sign_id == sign->id)) {
-      if ((sdict = sign_get_info(sign)) != NULL) {
-        tv_list_append_dict(l, sdict);
-      }
+      tv_list_append_dict(l, sign_get_info(sign));
```

Comment:
using local vars like `sdict` is good practice for e.g. debugging. Next time I would just leave it.